anacron/runjob.c

```diff
@@ -247,13 +254,38 @@ launch_mailer(job_rec *jr)
 	/* fdflags = fcntl(0, F_GETFL); fdflags &= ~O_APPEND; */
 	/* fcntl(0, F_SETFL, fdflags ); */
 
-	/* Here, I basically mirrored the way /usr/sbin/sendmail is called
-	 * by cron on a Debian system, except for the "-oem" and "-or0s"
-	 * options, which don't seem to be appropriate here.
-	 * Hopefully, this will keep all the MTAs happy. */
-	execl(SENDMAIL, SENDMAIL, "-FAnacron", "-odi",
-	      jr->mailto, (char *)NULL);
-	die_e("Can't exec " SENDMAIL);
+        /* Get SENDMAIL variable from the environment if set. 
+        /* If not, will use the predefined SENDMAIL from global.h */
+        mailer_full = getenv("SENDMAIL");
+        
+        if (mailer_full == NULL) {
+            /* Here, I basically mirrored the way /usr/sbin/sendmail is called
```

Comment:
Here I wouldn't duplicate the execl call but instead assign SENDMAIL to mailer_full if it is NULL.

Reply:
> Here I wouldn't duplicate the execl call but instead assign SENDMAIL to mailer_full if it is NULL.

I don't know how to do it, without making it into a special case. We need to pass these arguments to sendmail.